File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    stream_from 'chat:message'
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def post(data)
    ChatChannel.broadcast_to('message', data['message'])
  end
end

File: app/javascript/channels/chat_channel.js
import consumer from "./consumer"

window.chat = consumer.subscriptions.create("ChatChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    console.log(data)
  },

  post(message) {
    this.perform('post', { message: message })
  },
});
